logp/core.go

```diff
@@ -151,6 +151,56 @@ func ConfigureWithOutputs(defaultLoggerCfg Config, outputs ...zapcore.Core) erro
 	return nil
 }
 
+// ConfigureWithCore configures the global logger to use an output created
+// from `defaultLoggerCfg` and all the output passed by `output`.
+func ConfigureWithCore(defaultLoggerCfg Config, core zapcore.Core) error {
```

Comment:
@leehinman, I'm confused here, how is this function conceptually different from `ConfigureWithOutputs`?

Even the GoDoc for it is pretty much the same :P

Reply:
observedLogs and the default logger output.  But yes, these are very similar functions.  I'm fine with saying we should refactor.  The reason for opening the PR was to get you to look at it :-)